#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: fix man page include

--- doc/builtins.1~	Mon Nov 29 22:30:13 1999
+++ doc/builtins.1	Tue Aug  1 21:54:06 2000
@@ -10,6 +10,6 @@
 ulimit, umask, unalias, unset, until, wait, while \- bash built-in commands, see \fBbash\fR(1)
 .SH BASH BUILTIN COMMANDS
 .nr zZ 1
-.so bash.1
+.so /usr/share/man/man1/bash.1
 .SH SEE ALSO
 bash(1), sh(1)
